overlord/ifacestate/helpers.go

```diff
@@ -719,3 +726,41 @@ func ensureSystemSnapIsPresent(st *state.State) error {
 	_, err := snapstate.CoreInfo(st)
 	return err
 }
+
+func hotplugTaskSetAttrs(task *state.Task, deviceKey, ifaceName string) {
+	task.Set("device-key", deviceKey)
+	task.Set("interface", ifaceName)
+}
+
+func hotplugTaskGetAttrs(task *state.Task) (deviceKey, ifaceName string, err error) {
+	if err = task.Get("interface", &ifaceName); err != nil {
+		return "", "", fmt.Errorf("internal error: failed to get interface name: %s", err)
+	}
+	if err = task.Get("device-key", &deviceKey); err != nil {
+		return "", "", fmt.Errorf("internal error: failed to get device key: %s", err)
+	}
+	return deviceKey, ifaceName, err
+}
+
+type HotplugSlotDef struct {
```

Comment:
Why "Def"?  Doesn't sound like a convention we have in place, and in general every struct is a "*Def".

Reply:
Renamed to HotplugSlotInfo, per your suggestion from the followup PR, thanks!